src/components/sync-dialog/SyncDialog.jsx

```diff
@@ -61,7 +66,8 @@ class SyncDialog extends React.Component {
 
     render() {
         const { d2, isOpen } = this.props;
-
+        const { targetInstances } = this.state;
+        const disableSync = targetInstances && _.isEmpty(targetInstances);
```

Comment:
Minor: AFICS, `targetInstances` is always an array, so the first check can be removed.